src/server/auth/server/api_server.go

```diff
@@ -750,57 +747,30 @@ func (a *apiServer) ModifyAdmins(ctx context.Context, req *auth.ModifyAdminsRequ
 	// Canonicalize GitHub usernames in request (must canonicalize before we can
 	// validate, so we know who is actually being added/removed & can confirm
 	// that not all admins are being removed)
-	eg := &errgroup.Group{}
-	canonicalizedToAdd := make([]string, len(req.Add))
-	for i, user := range req.Add {
-		i, user := i, user
-		eg.Go(func() error {
-			user, err = a.canonicalizeSubject(ctx, user)
-			if err != nil {
-				return err
-			}
-			canonicalizedToAdd[i] = user
-			return nil
-		})
-	}
-	canonicalizedToRemove := make([]string, len(req.Remove))
-	for i, user := range req.Remove {
-		i, user := i, user
-		eg.Go(func() error {
-			user, err = a.canonicalizeSubject(ctx, user)
-			if err != nil {
-				return err
-			}
-			canonicalizedToRemove[i] = user
-			return nil
-		})
-	}
-	if err := eg.Wait(); err != nil {
+	canonicalizedUser, err := a.canonicalizeSubject(ctx, req.Principal)
+	if err != nil {
 		return nil, err
 	}
-	err = a.validateModifyAdminsRequest(canonicalizedToAdd, canonicalizedToRemove)
+
+	err = a.validateModifyAdminsRequest(canonicalizedUser, *req.Scopes)
 	if err != nil {
 		return nil, err
 	}
 
-	// Update "admins" list (watchAdmins() will update admins cache)
-	for _, user := range canonicalizedToAdd {
+	// Update "admins" list (watchAdmins() will update admins cache).
+	if len(req.Scopes.Scopes) > 0 {
```

Comment:
Probably obvious, but here (and in `validateModifyAdminsRequest` we should make sure to change logic a bit so that if a principal is being given the `NONE` admin scope, then they're deleted from the admin collection (I think currently here and in `validate...` is uses the old logic, which just writes the `NONE` scope to etcd)

Reply:
This code currently detects if there's no admin with `SUPER`, and removes the entry in etcd if the list of roles is empty. I think this misunderstanding is because `UNAVAILABLE` doesn't mean "remove all roles" in this PR, it's an invalid value. An empty list of roles means "remove all roles".

Reply:
Ah, sorry, I honestly just forgot that the enum was structured this way—this makes sense!